internal/keystore/vault/client.go

```diff
@@ -168,9 +168,14 @@ func (c *client) RenewToken(ctx context.Context, authenticate authFunc, secret *
 		}
 
 		// We renew the token right before it expires.
-		renewIn := ttl
-		if renewIn > 10*time.Second {
-			renewIn = ttl - 10*time.Second
+		var renewIn time.Duration
```

Comment:
I think a  better? code is to renew at 80% of the elpased time before the expiration. This can give us peace of mind

Reply:
We can also do that but this means for a TTL of 1h you renew it 12min before expiry. I would be okay with that but the next thing might be that KES is renewing the token "too early"...

Reply:
Too early is not a problem, too often is.

Reply:
One might hope but I've seen some wild security theature